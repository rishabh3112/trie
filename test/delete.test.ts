import Store from "../src";

describe("delete funtionality", () => {
  it("should delete single data item from store", async () => {
    const store = new Store<string>("test");

    await store.add("rishabh", "https://rishabhchawla.co");

    const beforeDelete = await store.at("rishabh");
    expect(beforeDelete).toMatchSnapshot();

    await store.delete("rishabh");

    try {
      await store.at("rishabh");
    } catch (err) {
      expect(err).toMatchSnapshot();
    }
  });

  it("should throw error if item doesn't exists", async () => {
    const store = new Store<string>("test");
    try {
      await store.delete("rishabh");
    } catch (err) {
      expect(err).toMatchSnapshot();
    }
  });
});
