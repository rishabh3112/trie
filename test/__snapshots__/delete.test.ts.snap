// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete funtionality should delete single data item from store 1`] = `
Object {
  "data": "https://rishabhchawla.co",
  "id": "test#7",
  "key": "rishabh",
}
`;

exports[`delete funtionality should delete single data item from store 2`] = `"Data doesn't exists"`;

exports[`delete funtionality should throw error if item doesn't exists 1`] = `"Data doesn't exists"`;
